warehouse/query-core/src/main/java/datawave/query/util/keyword/WordCloudResults.java

```diff
@@ -0,0 +1,172 @@
+package datawave.query.util.keyword;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import com.google.common.base.Objects;
+
+public class WordCloudResults {
+
+    final SortedSet<KeywordTuple> results = new TreeSet<>();
+    final Map<String,KeywordTuple> index = new HashMap<>();
+
+    public void addResults(KeywordResults results) {
+        final String source = results.getSource();
+        final String language = results.getLanguage();
+        final LinkedHashMap<String,Double> resultsMap = results.get();
+        for (Map.Entry<String,Double> e : resultsMap.entrySet()) {
+            KeywordTuple t = index.computeIfAbsent(e.getKey(), KeywordTuple::new);
+            t.addSourceScore(source, e.getValue(), language);
+        }
+    }
+
+    public SortedSet<KeywordTuple> finish() {
+        for (Map.Entry<String,KeywordTuple> e : index.entrySet()) {
+            results.add(e.getValue().finishScoring());
+        }
+        return results;
+    }
+
+    public String toString() {
```

Comment:
gson and jackson are both already dependencies of this module. Might be good to just use a library to generate the json?

Reply:
Yes, this is a good idea